### PR TITLE
Write out the output file even on internal exceptions.

### PR DESCRIPTION
**Pull request by mpacula**

### CODE REVIEW

File: bin/psprofile [hunk 1]
```diff
@@ -19,27 +20,33 @@ if __name__ == '__main__':
...
+            out_dir = os.path.dirname(args.output_file)
+            if not os.path.exists(out_dir):
+                raise IOError('output directory does not exist: `%s`' % out_dir)
+
+        # Run Profile
+        output = main(args.command, args.poll_interval, args.shell, args.skip_profile)
+        # Write output
+        output_json = json.dumps(output, indent=4)
+    except:
+        print >> sys.stderr, "Unexpected psprofile exit. Exception traceback below."
```
Comment (alliemclean) [inline bin/psprofile:46]: Not critical but why did you choose this syntax?

Comment (alliemclean) [reply bin/psprofile:46]: Oh, it's consistent with elsewhere I see.

File: bin/psprofile [hunk 2]
```diff
@@ -19,27 +20,33 @@ if __name__ == '__main__':
...
+        traceback.print_exc(file=sys.stderr)
+        output = {'exit_status': 1}
+        output_json = json.dumps(output, indent=4)
```
Comment (alliemclean) [inline bin/psprofile:49]: I could be mistaken but it looks to me like it will still exit with a failure status below.

File: bin/psprofile [hunk 3]
```diff
@@ -19,27 +20,33 @@ if __name__ == '__main__':
...
+        if args.output_file:
+            out_dir = os.path.dirname(args.output_file)
+            if not os.path.exists(out_dir):
+                raise IOError('output directory does not exist: `%s`' % out_dir)
+
+        # Run Profile
+        output = main(args.command, args.poll_interval, args.shell, args.skip_profile)
+        # Write output
+        output_json = json.dumps(output, indent=4)
+    except:
```
Comment (alliemclean) [inline bin/psprofile:45]: Should you only except the IOError?

**Review by alliemclean — APPROVED** [review]
Minor comments.

File: bin/psprofile (hunk 3 above)
Comment (mpacula) [reply bin/psprofile:45]: There may be other runtime errors thrown inside main(...) on line 42.

File: bin/psprofile (hunk 1 above)
Comment (mpacula) [reply bin/psprofile:46]: Yup.

**Pull request merged**